libstorage.apib

```diff
@@ -468,33 +531,50 @@ Internal server error
 
     + Schema
 
-            { "$ref": "https://raw.githubusercontent.com/emccode/libstorage/master/libstorage.json#/definitions/internalServerError" }
+            { "$ref": "https://raw.githubusercontent.com/codedellemc/libstorage/master/libstorage.json#/definitions/internalServerError" }
 
 ## Get with Attachments [GET /volumes?{attachments}]
 Gets a list of Volume resources and their attached resources for all configured
 services.
 
 + Parameters
 
-    + attachments (required)
-
-        A flag that indicates that volume attachments should be retrieved.
-        If this flag is provided then the `LibStorage-InstanceID` header must
-        be set -- a comma-delimited list of service name/instance ID,
-        colon-delimited pairs.
+    + attachments (string,optional)
+
+        A value that indicates how and if volume attachments should be retrieved.
+        <br/><br/>
+        The `attachments` parameter can be any of the following values:
+        <br/>
+        Value | Description
+        ------|------------
+        0 | No attachment information is requested. This is the default value and the same as omitting this parameter altogether.
+        1 | Attachment information is requested. This value indicates attachment information should be provided for all volumes.
+        2 | _My_ attachment information is requested. This value indicates attachment information should be provided for volumes attached to the instance specified in the [instance ID](#def-instance-id) request header should be returned. If this bit is set then the instance ID header is required.
+        4 | Attempt to map devices provided via the [local devices](#def-local-devices) request header to the appropriate attachment information. If this bit is set then the instance ID and local device headers are required.
```

Comment:
Hi @codenrhoden,

> But for GET /volumes, there isn't a specific service involved. So what does a client do in that case? Provide a full device map of everything attached?

Yes. When the libStorage client is first initialized it asks the server for information about all configured services. From that a unique set of drivers is compiled and for each driver the executor is used to grab the driver's instance ID. This is cached for the duration of the client.

For local devices, any call that might require local devices causes the executor to gather that information. For service-specific calls the local devices are retrieved only for the service's driver type. For global calls all local device maps are obtained -- one for each driver type.